### PR TITLE
[NCL-3816] Fix predicates for buildconfigset

### PR DESCRIPTION
**Pull request by thescouser89**
The predicate fix is for getting a list of BuildRecord from a build
config set id.

### CODE REVIEW

**Comment by thescouser89** [comment]
@matejonnet  @janinko @jbartece Because it *seems* I found an old bug from 2015, and I'm not even sure of myself, could you please verify this PR?

It seems there was confusion on build configuration set id and build configuration set record id for NCL-3816, but the commit was done in 2015 and I'm unsure how that bug would have been unnoticed for so long.

Could you also verify my predicate for: `withBuildConfigSetId`? I think I got it right but heh it's Dustin!

The Buildconfigsetrecord has a foreign key on buildconfigurationset. Once we obtain the buildconfigsetrecord, we can then get all the build records linking to those particular buildconfigsetrecord.

File: spi/src/main/java/org/jboss/pnc/spi/datastore/predicates/BuildRecordPredicates.java [hunk 1]
```diff
@@ -73,8 +75,12 @@
 
     public static Predicate<BuildRecord> withBuildConfigSetId(Integer buildConfigSetId) {
         return (root, query, cb) -> {
-            Join<BuildRecord, BuildConfigSetRecord> joinedConfiguSet = root.join(BuildRecord_.buildConfigSetRecord);
-            return cb.equal(joinedConfiguSet.get(org.jboss.pnc.model.BuildConfigSetRecord_.id), buildConfigSetId);
```
Comment (thescouser89) [inline spi/src/main/java/org/jboss/pnc/spi/datastore/predicates/BuildRecordPredicates.java]: Here we're confusing build config set id and build config set record id

**Comment by janinko** [comment]
Yeah, it seems weird that there were two methods doing the same. Looked at your change and couldn't find any obvious issue.

**Comment by thescouser89** [comment]
Merging it for further testing in our Devel environment

**Pull request merged**